Property 'sessionPlugin' is missing in type '{ cookieOpts: { signed: true; }; }' but required in type 'FastifyCsrfProtectionOptionsFastifySecureSession'

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Fastify version

8.2.1

### Plugin version

6.3.0

### Node.js version

16.13.2

### Operating system

Windows

### Operating system version (i.e. 20.04, 11.3, 10)

11

### Description

```
app.register(fastifyCsrfProtection, { cookieOpts: { signed: true } }) 
```

Typescript is throwing the following error:

```
Argument of type '{ cookieOpts: { signed: true; }; }' is not assignable to parameter of type 'FastifyRegisterOptions<FastifyCsrfProtectionOptions>'.
  Type '{ cookieOpts: { signed: true; }; }' is not assignable to type 'RegisterOptions & FastifyCsrfProtectionOptionsBase & FastifyCsrfProtectionOptionsFastifySecureSession'.
    Property 'sessionPlugin' is missing in type '{ cookieOpts: { signed: true; }; }' but required in type 'FastifyCsrfProtectionOptionsFastifySecureSession'.
```

### Steps to Reproduce

prepare a project with NestJs and Fastify. then use the @fastify/cookie with @fastify/csrf-protection.

in the mean file include:

```
     app.register(fastifyCookie, {
        secret: configService.get<string>('COOKIE_SECRET')
    })

    app.register(fastifyHelmet)

    app.register(fastifyCsrfProtection, { cookieOpts: { signed: true } })
```


### Expected Behavior

not to see such an error. according to you the documentation adding `{ cookieOpts: { signed: true } }` as second parameter should be sufficient.

FastifyCsrfProtectionOptionsFastifySecureSession is defined in `fastifyCsrfProtection` . as the following:

```
interface FastifyCsrfProtectionOptionsFastifySecureSession {
    sessionPlugin: '@fastify/secure-session';
    csrfOpts?: CSRFOptions;
  }
```

as sessionPlugin has already a default value of  '@fastify/secure-session'. shouldn't it be optional in the definition ??

This isn't strictly related to NestJS, but to the Typescript types. [According to the README](https://github.com/fastify/csrf-protection#use-with-fastifycookie) it's possible to use the plugin without passing a `sessionPlugin` option to the `register`'s second parameter. However, [the types](https://github.com/fastify/csrf-protection/blob/master/types/index.d.ts#L53) say that the `sessionPlugin` property is required and should be `'@fastify/session'`, `@fastify/session-secure'` or `'@fastify/cookie'`. Now, when adding `sessionPlugin: '@fastify/cookie'`, there's a Typescript error
```
src/main.ts:48:5 - error TS2322: Type '"@fastify/cookie"' is not assignable to type '"@fastify/secure-session"'.

48     sessionPlugin: "@fastify/cookie",
       ~~~~~~~~~~~~~

[2:14:16 PM] Found 1 error. Watching for file changes.
```

I'll put together a a minimal non-NestJS reproduction in a moment. [I found this after providing some support to a StackOverflow Question](https://stackoverflow.com/q/76154835/9576186)

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -36,18 +36,16 @@ declare namespace fastifyCsrfProtection {
     getToken?: GetTokenFn;
   }
 
+  type RequiredCSRFHmac = Omit<CSRFOptions, 'hmacKey' | 'userInfo'> &
+    Required<Pick<CSRFOptions, 'hmacKey'>> & { userInfo: true }
+
+  type OptionalCSRFNoUserInfo = Omit<CSRFOptions, 'userInfo'> & {
+    userInfo?: false;
+  }
+
   interface FastifyCsrfProtectionOptionsFastifyCookie {
     sessionPlugin?: '@fastify/cookie';
-    csrfOpts?: | ({
-      [k in keyof CSRFOptions]: k extends 'userInfo'
-        ? true
-        : CSRFOptions[k];
-    } & Required<Pick<CSRFOptions, 'hmacKey'>>)
-  | ({
-    [k in keyof CSRFOptions]: k extends 'userInfo'
-      ? false
-      : CSRFOptions[k];
-  });
+    csrfOpts?: OptionalCSRFNoUserInfo | RequiredCSRFHmac;
   }
 
   interface FastifyCsrfProtectionOptionsFastifySession {

